clusterloader/cluster_loader.go

```diff
@@ -0,0 +1,191 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cl
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"time"
+
+	clframe "github.com/kubernetes/perf-tests/clusterloader/framework"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"k8s.io/kubernetes/pkg/api/v1"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/framework"
+	testutils "k8s.io/kubernetes/test/utils"
+)
+
+var _ = framework.KubeDescribe("Cluster Loader [Performance] [Slow] [Disruptive]", func() {
+	f := framework.NewDefaultFramework("cluster-loader")
+	defer ginkgo.GinkgoRecover()
+
+	var c clientset.Interface
+	ginkgo.BeforeEach(func() {
+		c = f.ClientSet
+	})
+
+	ginkgo.It(fmt.Sprintf("running config file"), func() {
+		project := clframe.ConfigContext.ClusterLoader.Projects
+		tuningSets := clframe.ConfigContext.ClusterLoader.TuningSets
+		if len(project) < 1 {
+			framework.Failf("invalid config file.\nFile: %v", project)
+		}
+
+		var namespaces []*v1.Namespace
+		//totalPods := 0 // Keep track of how many pods for stepping
+		// TODO sjug: add concurrency
+		for _, p := range project {
+			// Find tuning if we have it
+			tuning := getTuningSet(tuningSets, p.Tuning)
+			framework.Logf("Our tuning set is: %v", tuning)
+			for j := 0; j < p.Number; j++ {
+				// Create namespaces as defined in Cluster Loader config
+				nsName := appendIntToString(p.Basename, j)
+				ns, err := f.CreateNamespace(nsName, nil)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				framework.Logf("%d/%d : Created new namespace: %v", j+1, p.Number, nsName)
+				namespaces = append(namespaces, ns)
+
+				// Create templates as defined
+				for _, v := range p.Templates {
+					createTemplate(v.Basename, ns, mkPath(v.File), v.Number, tuning)
+				}
+				// This is too familiar, create pods
+				for _, v := range p.Pods {
+					config := parsePods(mkPath(v.File))
+					clframe.CreatePods(f, v.Basename, ns.Name, config.Spec, v.Number, tuning)
+				}
+			}
+		}
+
+		// Wait for pods to be running
+		for _, ns := range namespaces {
+			label := labels.SelectorFromSet(labels.Set(map[string]string{"purpose": "test"}))
+			err := testutils.WaitForPodsWithLabelRunning(c, ns.Name, label)
+			if err != nil {
+				framework.Failf("Got %v when trying to wait for the pods to start", err)
+			}
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			framework.Logf("All pods running in namespace %s.", ns.Name)
+		}
+	})
+})
+
+// getTuningSet matches the name of the tuning set defined in the project and returns a pointer to the set
+func getTuningSet(tuningSets []clframe.TuningSetType, podTuning string) (tuning *clframe.TuningSetType) {
```

Comment:
We could but then we would potentially have to redefine the same tuningset for each project as opposed to just referencing the name of the same tuningset?